agent.go

```diff
@@ -1210,3 +1221,59 @@ func winsizeCb(pod *pod, data []byte) error {
 
 	return nil
 }
+
+//Get the agent configuration from kernel cmdline
+func getConfig() agentConfig {
+
+	config := agentConfig{}
+	config.logLevel = defaultLogLevel
+
+	kernelCmdline, err := ioutil.ReadFile(kernelCmdlineFile)
+	if err != nil {
+		agentLog.Warnf("Failed to read kernel cmdline %v", err)
+		return config
+	}
+
+	words := strings.Fields(string(kernelCmdline))
+	for _, w := range words {
+		word := string(w)
+		if err := config.parseCmdlineOption(word); err != nil {
+			agentLog.Warnf("Failed to parse kernel option %s: %v", word, err)
+		}
+	}
+	return config
+}
+
+func applyConfig(config agentConfig) {
+	agentLog.SetLevel(config.logLevel)
+}
+
+//Parse a string that represents a kernel cmdline option
+func (c *agentConfig) parseCmdlineOption(option string) error {
+
+	const (
+		optionPosition = iota
+		valuePosition
+		optionSeparator = "="
+	)
+
+	split := strings.Split(option, optionSeparator)
+
+	if len(split) < valuePosition+1 {
+		return nil
+	}
+
+	option = split[optionPosition]
```

Comment:
Don't assign the key that you parsed to the option, instead, use the key (that is `split[optionPosition]`) directly into the switch:
```go
switch split[optionPosition] {
    ...
```